usr/src/cmd/zfs/zfs_main.c

```diff
@@ -3769,15 +3769,42 @@ zfs_do_send(int argc, char **argv)
 			flags.compress = B_TRUE;
 			break;
 		case ':':
-			(void) fprintf(stderr, gettext("missing argument for "
-			    "'%c' option\n"), optopt);
+			/*
+			 * If a parameter was not passed, optopt contains the
+			 * value that would normally lead us into the
+			 * appropriate case statement.  If it's > 256, then this
+			 * must be a longopt and we should look at argv to get
+			 * the string.  Otherwise it's just the character, so we
+			 * should use it directly.
+			 */
+			if (optopt <= UINT8_MAX) {
```

Comment:
In the bug report you talk about this being the value in the fourth column of getopt. How does this translate into a value that's greater than or less than 256? Did you just pick 256 because of it being 8-bits or is this supposed to be trying to relate to ascii or something else? All of our long options today are 7-bit ascii characters.